.github/workflows/agent-build-new.yml

```diff
@@ -39,14 +39,13 @@ jobs:
           github_token: ${{ github.token }}
 
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
 
       - id: define_constants
         run: |
           echo "python_version=3.8" >> "${GITHUB_OUTPUT}"
-          echo "cache_version=6" >> "${GITHUB_OUTPUT}"
+          echo "cache_version=$GITHUB_REF_NAME" >> "${GITHUB_OUTPUT}"
```

Comment:
This means we won't be able to reuse cache across branches since this variable contains branch name, right?

Reply:
Yes. What I found was that some tests were passing only because of the cache (from a different branch perhaps?) and hiding and error. So I wanted to restrict it.

Reply:
Do we know if it will have significant impact on test run time?

But otherwise I think that's fine and similar approach we use in other places (re-build everything from scratch in master branch).

Reply:
The build and test time is still up to 20 minutes.